### PR TITLE
[tests-only][full-ci]added test to list shares sharedWithMe after the share role has been disabled

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds the test to list shares `sharedWithMe` after the share role has been disabled, with precondition 
`enable role` -> `share` -> `disable role`

## Related Issue

- https://github.com/owncloud/ocis/issues/10711

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature [hunk 1]
```diff
@@ -3916,10 +3916,83 @@ Feature: resources shared by user
...
     Examples:
-      | resource      |
-      | textfile.txt  |
-      | folderToShare |
+      | resource      | resource-type |
+      | textfile.txt  | file          |
+      | folderToShare | folder        |
 
   @env-config
   Scenario: sharer lists the shares shared with Denied after the role is disabled (Personal Space)
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3998]:
```suggestion
  Scenario: users list the shares shared with Denied after the role is disabled (Personal Space)
```

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature [hunk 2]
```diff
@@ -3916,13 +3916,86 @@ Feature: resources shared by user
...
-      | resource      |
-      | textfile.txt  |
-      | folderToShare |
+      | resource      | resource-type |
+      | textfile.txt  | file          |
+      | folderToShare | folder        |
 
   @env-config
-  Scenario: sharer lists the shares shared with Denied after the role is disabled (Personal Space)
+  Scenario: user list the shares shared with Denied after the role is disabled (Personal Space)
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3998]:
```suggestion
  Scenario: users list the shares shared with Denied after the role is disabled (Personal Space)
```

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature [hunk 3]
```diff
@@ -3865,7 +3865,7 @@ Feature: resources shared by user
       """
 
   @env-config
-  Scenario Outline: sharer lists the shares shared with Secure Viewer after the role is disabled (Personal Space)
+  Scenario Outline: user list the shares shared with Secure Viewer after the role is disabled (Personal Space)
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3868]: i think its best if we use the words sharer and sharee in case of shares

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature [hunk 4]
```diff
@@ -3916,13 +3916,86 @@ Feature: resources shared by user
         }
       }
       """
+    When user "Brian" lists the shares shared with him using the Graph API
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3921]: its interesting you are mixing sharebyme and sharewithme tests. 
@saw-jan are you ok with this.
tests/acceptance/features/apiSharingNg1/sharedWithMe.feature this file is dedicated to share-with-me 
I vote not to mix those if we are mixing then mix others also because it will only create confusion in future

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3868]: users refers to both sharer and sharee

File: tests/acceptance/features/apiSharingNg1/sharedByMe.feature (hunk 4 above)
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3921]: let's move this kind of tests to own feature file (enable/disable role). this file is already huge

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg1/sharedByMe.feature:3921]: Previously it was discussed to have it covered in single scenario (https://github.com/owncloud/ocis/pull/10752#discussion_r1881351989). Yeah we can have a separate feature file for these scenarios to prevent the confusion. :+1:

File: tests/acceptance/features/apiSharingNg1/enableDisablePermissionsRole.feature [hunk 5]
```diff
@@ -0,0 +1,401 @@
+@env-config
+Feature: Enable disable permissions role
```
Comment (prashant-gurung899) [inline tests/acceptance/features/apiSharingNg1/enableDisablePermissionsRole.feature:2]:
```suggestion
Feature: enable/disable permissions role
```

File: tests/acceptance/features/apiSharingNg1/enableDisablePermissionsRole.feature [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+  Background:
+    Given these users have been created with default attributes:
+      | username |
+      | Alice    |
+      | Brian    |
+
+
+  Scenario Outline: users list the shares shared with Secure Viewer after the role is disabled (Personal Space)
+    Given the administrator has enabled the permissions role "Secure Viewer"
+    And user "Alice" has uploaded file with content "some content" to "textfile.txt"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg1/enableDisablePermissionsRole.feature:16]: maybe we should assert these in a single scenario for these kinds of tests (edge-cases):
- sharedByMe (sharer)
- sharedWithMe (sharee)
- list me/drives (sharee)
- PROPFIND Shares space (sharee)
- some action by sharee (e.g.: download file)

Comment (PrajwolAmatya) [reply tests/acceptance/features/apiSharingNg1/enableDisablePermissionsRole.feature:16]: I have updated the changes in the follow up PR https://github.com/owncloud/ocis/pull/10756. We can merge this PR and add these changes in other PR.

**Review by saw-jan — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]
